// @angular/material/core/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/core';
import * as i1 from '@angular/platform-browser';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i4 from '@angular/core';
import * as i5 from '@angular/cdk/platform';
import * as i6 from '@angular/cdk/scrolling';
import * as i7 from '@angular/common';
import * as i8 from '../../../cdk/platform/typings/index.ngsummary';
import * as i9 from '../../../cdk/scrolling/typings/index.ngsummary';
import * as i10 from '../../../common/common.ngsummary';
import * as i11 from '@angular/cdk/a11y';
import * as i12 from '../../../cdk/a11y/typings/index.ngsummary';
import * as i13 from './index.ngfactory';
export function MdCommonModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdCommonModule,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DOCUMENT}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i2.DIR_DOCUMENT}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i1.DOCUMENT}},deps:(undefined as any),
          multi:false},module:{reference:i2.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i2.Directionality,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i2.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i2.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i0.MdCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}}],modules:[{reference:i0.CompatibilityModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i2.BidiModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.MdCommonModule,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
          lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i0.MatPrefixRejector},
          {reference:i0.MdPrefixRejector},{reference:i2.Dir}],exportedPipes:([] as any[])},
      CompatibilityModuleNgSummary,i3.BidiModuleNgSummary];
}
export function CompatibilityModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.CompatibilityModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:([] as any[]),
      modules:[{reference:i0.CompatibilityModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MatPrefixRejector},{reference:i0.MdPrefixRejector}],
      exportedPipes:([] as any[])},MatPrefixRejectorNgSummary,MdPrefixRejectorNgSummary];
}
export function NoConflictStyleCompatibilityModeNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.NoConflictStyleCompatibilityMode,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i0.MATERIAL_COMPATIBILITY_MODE}},
      useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
      deps:(undefined as any),multi:false},module:{reference:i0.NoConflictStyleCompatibilityMode,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i0.NoConflictStyleCompatibilityMode,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),
      exportedPipes:([] as any[])}];
}
export function NativeDateModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.NativeDateModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i0.DateAdapter}},
      useClass:{reference:i0.NativeDateAdapter,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],
      multi:false},module:{reference:i0.NativeDateModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i0.MAT_DATE_LOCALE}},useClass:(null as any),
          useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i4.LOCALE_ID}},
          deps:(undefined as any),multi:false},module:{reference:i0.NativeDateModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i0.NativeDateModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),
      exportedPipes:([] as any[])},{summaryKind:3,type:{reference:i0.NativeDateAdapter,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],lifecycleHooks:([] as any[])}}];
}
export function MdNativeDateModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdNativeDateModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i0.DateAdapter}},
      useClass:{reference:i0.NativeDateAdapter,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],
      multi:false},module:{reference:i0.NativeDateModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i0.MAT_DATE_LOCALE}},useClass:(null as any),
          useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i4.LOCALE_ID}},
          deps:(undefined as any),multi:false},module:{reference:i0.NativeDateModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.MD_DATE_FORMATS}},
          useClass:(null as any),useValue:i0.MAT_NATIVE_DATE_FORMATS,useFactory:(null as any),
          useExisting:(undefined as any),deps:(undefined as any),multi:false},module:{reference:i0.MdNativeDateModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i0.NativeDateModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.MdNativeDateModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),
      exportedPipes:([] as any[])},NativeDateModuleNgSummary];
}
export function MdLineModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdLineModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i2.DIR_DOCUMENT}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i1.DOCUMENT}},deps:(undefined as any),
          multi:false},module:{reference:i2.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i2.Directionality,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i2.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i2.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i0.MdCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}}],modules:[{reference:i0.CompatibilityModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i2.BidiModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.MdCommonModule,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
          lifecycleHooks:([] as any[])},{reference:i0.MdLineModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i0.MatPrefixRejector},
          {reference:i0.MdPrefixRejector},{reference:i2.Dir},{reference:i0.MdLine}],
      exportedPipes:([] as any[])},MdLineNgSummary,CompatibilityModuleNgSummary,i3.BidiModuleNgSummary,
      MdCommonModuleNgSummary];
}
export function MdOptionModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdOptionModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i2.DIR_DOCUMENT}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i1.DOCUMENT}},deps:(undefined as any),
          multi:false},module:{reference:i2.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i2.Directionality,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i2.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i2.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i0.MdCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i5.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i5.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
              useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
              multi:false},module:{reference:i5.PlatformModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i6.ScrollDispatcher}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i6.SCROLL_DISPATCHER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i6.ScrollDispatcher}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i4.NgZone}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i5.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i6.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i4.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.Platform}}}],multi:false},module:{reference:i6.ScrollDispatchModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i6.ViewportRuler}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i6.VIEWPORT_RULER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i6.ViewportRuler}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.ScrollDispatcher}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i6.ViewportRuler}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.ScrollDispatcher}}}],multi:false},
              module:{reference:i0.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i7.NgLocalization}},useClass:{reference:i7.NgLocaleLocalization,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i4.LOCALE_ID}}}],lifecycleHooks:([] as any[])},
              useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
              deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i4.LOCALE_ID}}}],multi:false},
              module:{reference:i7.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],
      modules:[{reference:i0.CompatibilityModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i2.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MdCommonModule,diDeps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DOCUMENT}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                  token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
          {reference:i5.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i6.ScrollDispatchModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i7.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MdPseudoCheckboxModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MdOptionModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MdOption},{reference:i0.MdOptgroup}],exportedPipes:([] as any[])},
      MdOptionNgSummary,MdOptgroupNgSummary,CompatibilityModuleNgSummary,i3.BidiModuleNgSummary,
      MdCommonModuleNgSummary,i8.PlatformModuleNgSummary,i9.ScrollDispatchModuleNgSummary,
      MdRippleModuleNgSummary,i10.CommonModuleNgSummary,MdPseudoCheckboxModuleNgSummary];
}
export function MdRippleModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i2.DIR_DOCUMENT}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i1.DOCUMENT}},deps:(undefined as any),
          multi:false},module:{reference:i2.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i2.Directionality,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i2.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i2.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i2.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i0.MdCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i5.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i5.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
              useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
              multi:false},module:{reference:i5.PlatformModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i6.ScrollDispatcher}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i6.SCROLL_DISPATCHER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i6.ScrollDispatcher}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i4.NgZone}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i5.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i6.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i4.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.Platform}}}],multi:false},module:{reference:i6.ScrollDispatchModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i6.ViewportRuler}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i6.VIEWPORT_RULER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i6.ViewportRuler}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.ScrollDispatcher}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i6.ViewportRuler}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.ScrollDispatcher}}}],multi:false},
              module:{reference:i0.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],
      modules:[{reference:i0.CompatibilityModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i2.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MdCommonModule,diDeps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DOCUMENT}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                  token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
          {reference:i5.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i6.ScrollDispatchModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MatPrefixRejector},{reference:i0.MdPrefixRejector},
          {reference:i2.Dir},{reference:i0.MdRipple}],exportedPipes:([] as any[])},
      MdRippleNgSummary,CompatibilityModuleNgSummary,i3.BidiModuleNgSummary,MdCommonModuleNgSummary,
      i8.PlatformModuleNgSummary,i9.ScrollDispatchModuleNgSummary];
}
export function MdPseudoCheckboxModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdPseudoCheckboxModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:([] as any[]),
      modules:[{reference:i0.MdPseudoCheckboxModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MdPseudoCheckbox}],exportedPipes:([] as any[])},
      MdPseudoCheckboxNgSummary];
}
export function StyleModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.StyleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i7.NgLocalization}},
          useClass:{reference:i7.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.LOCALE_ID}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.LOCALE_ID}}}],
          multi:false},module:{reference:i7.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i5.Platform,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},useClass:{reference:i5.Platform,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:([] as any[]),multi:false},module:{reference:i5.PlatformModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i11.InteractivityChecker,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i5.Platform}}}],lifecycleHooks:([] as any[])}},
              useClass:{reference:i11.InteractivityChecker,diDeps:[{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.Platform}}}],
                  lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.Platform}}}],
              multi:false},module:{reference:i11.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i11.FocusTrapFactory,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i11.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i5.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.NgZone}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i11.FocusTrapFactory,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i11.InteractivityChecker}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.Platform}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i4.NgZone}}}],lifecycleHooks:([] as any[])},
              useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
              deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i11.InteractivityChecker}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.Platform}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i4.NgZone}}}],multi:false},module:{reference:i11.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i11.AriaDescriber,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i5.Platform}}}],lifecycleHooks:[1]}},
              useClass:{reference:i11.AriaDescriber,diDeps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.Platform}}}],
                  lifecycleHooks:[1]},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.Platform}}}],
              multi:false},module:{reference:i11.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i11.LiveAnnouncer}},useClass:(null as any),
              useValue:(undefined as any),useFactory:{reference:i11.LIVE_ANNOUNCER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i11.LiveAnnouncer}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:true,token:{identifier:{reference:i11.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i5.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i11.LiveAnnouncer}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i11.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.Platform}}}],multi:false},module:{reference:i11.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i11.AriaDescriber}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i11.ARIA_DESCRIBER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i11.AriaDescriber}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i5.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i11.AriaDescriber}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.Platform}}}],multi:false},module:{reference:i11.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i11.FocusMonitor}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i11.FOCUS_MONITOR_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i11.FocusMonitor}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i4.NgZone}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i5.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i11.FocusMonitor}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i4.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.Platform}}}],multi:false},module:{reference:i11.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i7.CommonModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i5.PlatformModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i11.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.StyleModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i11.FocusTrapDirective},
          {reference:i11.FocusTrapDeprecatedDirective},{reference:i11.CdkMonitorFocus}],
      exportedPipes:([] as any[])},i10.CommonModuleNgSummary,i8.PlatformModuleNgSummary,
      i12.A11yModuleNgSummary];
}
export function MatPrefixRejectorNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatPrefixRejector,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MATERIAL_COMPATIBILITY_MODE}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i4.ElementRef}}}],lifecycleHooks:([] as any[])},
      isComponent:false,selector:'\n  [mat-button],\n  [mat-fab],\n  [mat-icon-button],\n  [mat-mini-fab],\n  [mat-raised-button],\n  [matCardSubtitle],\n  [matCardTitle],\n  [matCellDef],\n  [matColumnDef],\n  [matDialogActions],\n  [matDialogClose],\n  [matDialogContent],\n  [matDialogTitle],\n  [matHeaderCellDef],\n  [matHeaderRowDef],\n  [matLine],\n  [matRowDef],\n  [matStepLabel],\n  [matStepperNext],\n  [matStepperPrevious],\n  [matTabLabel],\n  [matTabLink],\n  [matTabNav],\n  [matTooltip],\n  [matInput],\n  [matPrefix],\n  [matSuffix],\n  mat-autocomplete,\n  mat-button-toggle,\n  mat-button-toggle,\n  mat-button-toggle-group,\n  mat-card,\n  mat-card-actions,\n  mat-card-content,\n  mat-card-footer,\n  mat-card-header,\n  mat-card-subtitle,\n  mat-card-title,\n  mat-card-title-group,\n  mat-cell,\n  mat-checkbox,\n  mat-chip,\n  mat-dialog-actions,\n  mat-dialog-container,\n  mat-dialog-content,\n  mat-divider,\n  mat-error,\n  mat-grid-list,\n  mat-grid-tile,\n  mat-grid-tile-footer,\n  mat-grid-tile-header,\n  mat-header-cell,\n  mat-header-row,\n  mat-hint,\n  mat-horizontal-stepper,\n  mat-icon,\n  mat-input-container,\n  mat-form-field,\n  mat-list,\n  mat-list-item,\n  mat-menu,\n  mat-nav-list,\n  mat-option,\n  mat-placeholder,\n  mat-progress-bar,\n  mat-pseudo-checkbox,\n  mat-radio-button,\n  mat-radio-group,\n  mat-row,\n  mat-select,\n  mat-sidenav,\n  mat-sidenav-container,\n  mat-slider,\n  mat-spinner,\n  mat-step,\n  mat-tab,\n  mat-table,\n  mat-tab-group,\n  mat-toolbar,\n  mat-vertical-stepper',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MdPrefixRejectorNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdPrefixRejector,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MATERIAL_COMPATIBILITY_MODE}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i4.ElementRef}}}],lifecycleHooks:([] as any[])},
      isComponent:false,selector:'\n  [md-button],\n  [md-fab],\n  [md-icon-button],\n  [md-mini-fab],\n  [md-raised-button],\n  [mdCardSubtitle],\n  [mdCardTitle],\n  [mdCellDef],\n  [mdColumnDef],\n  [mdDialogActions],\n  [mdDialogClose],\n  [mdDialogContent],\n  [mdDialogTitle],\n  [mdHeaderCellDef],\n  [mdHeaderRowDef],\n  [mdLine],\n  [mdRowDef],\n  [mdStepLabel],\n  [mdStepperNext],\n  [mdStepperPrevious],\n  [mdTabLabel],\n  [mdTabLink],\n  [mdTabNav],\n  [mdTooltip],\n  [mdInput],\n  [mdPrefix],\n  [mdSuffix],\n  md-autocomplete,\n  md-button-toggle,\n  md-button-toggle,\n  md-button-toggle-group,\n  md-card,\n  md-card-actions,\n  md-card-content,\n  md-card-footer,\n  md-card-header,\n  md-card-subtitle,\n  md-card-title,\n  md-card-title-group,\n  md-cell,\n  md-checkbox,\n  md-chip,\n  md-dialog-actions,\n  md-dialog-container,\n  md-dialog-content,\n  md-divider,\n  md-error,\n  md-grid-list,\n  md-grid-tile,\n  md-grid-tile-footer,\n  md-grid-tile-header,\n  md-header-cell,\n  md-header-row,\n  md-hint,\n  md-horizontal-stepper,\n  md-icon,\n  md-input-container,\n  md-form-field,\n  md-list,\n  md-list-item,\n  md-menu,\n  md-nav-list,\n  md-option,\n  md-placeholder,\n  md-progress-bar,\n  md-pseudo-checkbox,\n  md-radio-button,\n  md-radio-group,\n  md-row,\n  md-select,\n  md-sidenav,\n  md-sidenav-container,\n  md-slider,\n  md-spinner,\n  md-step,\n  md-tab,\n  md-table,\n  md-tab-group,\n  md-toolbar,\n  md-vertical-stepper',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MdLineNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdLine,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[md-line], [mat-line], [mdLine], [matLine]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-line'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdOptionNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdOption,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i4.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MdOptgroup}}}],
      lifecycleHooks:([] as any[])},isComponent:true,selector:'md-option, mat-option',
      exportAs:(null as any),inputs:{value:'value',disabled:'disabled'},outputs:{onSelectionChange:'onSelectionChange'},
      hostListeners:{click:'_selectViaInteraction()',keydown:'_handleKeydown($event)'},
      hostProperties:{'attr.tabindex':'_getTabIndex()','class.mat-selected':'selected',
          'class.mat-option-multiple':'multiple','class.mat-active':'active',id:'id',
          'attr.aria-selected':'selected.toString()','attr.aria-disabled':'disabled.toString()',
          'class.mat-option-disabled':'disabled'},hostAttributes:{role:'option',class:'mat-option'},
      providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i0.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i13.View_MdOption_0,
      rendererType:i13.RenderType_MdOption,componentFactory:i13.MdOptionNgFactory}];
}
export function MdOptgroupNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdOptgroup,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'md-optgroup, mat-optgroup',exportAs:(null as any),
      inputs:{disabled:'disabled',label:'label'},outputs:{},hostListeners:{},hostProperties:{'class.mat-optgroup-disabled':'disabled',
          'attr.aria-disabled':'disabled.toString()','attr.aria-labelledby':'_labelId'},
      hostAttributes:{class:'mat-optgroup',role:'group'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:['md-option, mat-option'],
          encapsulation:2},componentViewType:i13.View_MdOptgroup_0,rendererType:i13.RenderType_MdOptgroup,
      componentFactory:i13.MdOptgroupNgFactory}];
}
export function MdRippleNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdRipple,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i4.NgZone}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.ViewportRuler}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i5.Platform}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MD_RIPPLE_GLOBAL_OPTIONS}}}],
      lifecycleHooks:[1,3]},isComponent:false,selector:'[md-ripple], [mat-ripple], [mdRipple], [matRipple]',
      exportAs:'mdRipple, matRipple',inputs:{trigger:'mdRippleTrigger',centered:'mdRippleCentered',
          disabled:'mdRippleDisabled',radius:'mdRippleRadius',speedFactor:'mdRippleSpeedFactor',
          color:'mdRippleColor',unbounded:'mdRippleUnbounded',_matRippleTrigger:'matRippleTrigger',
          _matRippleCentered:'matRippleCentered',_matRippleDisabled:'matRippleDisabled',
          _matRippleRadius:'matRippleRadius',_matRippleSpeedFactor:'matRippleSpeedFactor',
          _matRippleColor:'matRippleColor',_matRippleUnbounded:'matRippleUnbounded'},
      outputs:{},hostListeners:{},hostProperties:{'class.mat-ripple-unbounded':'unbounded'},
      hostAttributes:{class:'mat-ripple'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdPseudoCheckboxNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdPseudoCheckbox,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:true,selector:'md-pseudo-checkbox, mat-pseudo-checkbox',
      exportAs:(null as any),inputs:{state:'state',disabled:'disabled'},outputs:{},
      hostListeners:{},hostProperties:{'class.mat-pseudo-checkbox-indeterminate':'state === "indeterminate"',
          'class.mat-pseudo-checkbox-checked':'state === "checked"','class.mat-pseudo-checkbox-disabled':'disabled'},
      hostAttributes:{class:'mat-pseudo-checkbox'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:([] as any[]),
          encapsulation:2},componentViewType:i13.View_MdPseudoCheckbox_0,rendererType:i13.RenderType_MdPseudoCheckbox,
      componentFactory:i13.MdPseudoCheckboxNgFactory}];
}
export function UniqueSelectionDispatcherNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.UniqueSelectionDispatcher,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}}];
}
export function NativeDateAdapterNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.NativeDateAdapter,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],
      lifecycleHooks:([] as any[])}}];
}
export function GestureConfigNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.GestureConfig,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
