// @angular/material/tabs/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/tabs';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '@angular/cdk/bidi';
import * as i4 from '@angular/platform-browser';
import * as i5 from '@angular/material/core';
import * as i6 from '@angular/cdk/platform';
import * as i7 from '@angular/cdk/scrolling';
import * as i8 from '@angular/cdk/observers';
import * as i9 from '@angular/cdk/portal';
import * as i10 from '../../../common/common.ngsummary';
import * as i11 from '../../core/typings/index.ngsummary';
import * as i12 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i13 from '../../../cdk/portal/typings/index.ngsummary';
import * as i14 from '../../../cdk/platform/typings/index.ngsummary';
import * as i15 from '../../../cdk/scrolling/typings/index.ngsummary';
import * as i16 from '../../../cdk/observers/typings/index.ngsummary';
import * as i17 from './index.ngfactory';
export function MdTabsModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdTabsModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
          useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i3.DIR_DOCUMENT}},useClass:(null as any),
              useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i4.DOCUMENT}},
              deps:(undefined as any),multi:false},module:{reference:i3.BidiModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i3.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i3.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i3.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i5.MdCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i4.DOCUMENT}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i6.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i6.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
              useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
              multi:false},module:{reference:i6.PlatformModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.ScrollDispatcher}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i7.SCROLL_DISPATCHER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i7.ScrollDispatcher}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i2.NgZone}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i7.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i7.ScrollDispatchModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.ViewportRuler}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i7.VIEWPORT_RULER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i7.ViewportRuler}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i7.ScrollDispatcher}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i7.ViewportRuler}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i7.ScrollDispatcher}}}],multi:false},
              module:{reference:i5.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i8.MdMutationObserverFactory,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},useClass:{reference:i8.MdMutationObserverFactory,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
              useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
              multi:false},module:{reference:i8.ObserversModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.ViewportRuler}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i7.VIEWPORT_RULER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i7.ViewportRuler}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i7.ScrollDispatcher}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i7.ViewportRuler}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i7.ScrollDispatcher}}}],multi:false},
              module:{reference:i0.MdTabsModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],
      modules:[{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i5.CompatibilityModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i3.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i5.MdCommonModule,diDeps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:true,token:{identifier:{reference:i4.DOCUMENT}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                  token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
          {reference:i9.PortalModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i6.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i7.ScrollDispatchModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i5.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i8.ObserversModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MdTabsModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i5.MatPrefixRejector},{reference:i5.MdPrefixRejector},
          {reference:i3.Dir},{reference:i0.MdTabGroup},{reference:i0.MdTabLabel},{reference:i0.MdTab},
          {reference:i0.MdTabNav},{reference:i0.MdTabLink}],exportedPipes:([] as any[])},
      MdTabGroupNgSummary,MdTabLabelNgSummary,MdTabNgSummary,MdInkBarNgSummary,MdTabLabelWrapperNgSummary,
      MdTabNavNgSummary,MdTabLinkNgSummary,MdTabBodyNgSummary,MdTabHeaderNgSummary,
      i10.CommonModuleNgSummary,i11.CompatibilityModuleNgSummary,i12.BidiModuleNgSummary,
      i11.MdCommonModuleNgSummary,i13.PortalModuleNgSummary,i14.PlatformModuleNgSummary,
      i15.ScrollDispatchModuleNgSummary,i11.MdRippleModuleNgSummary,i16.ObserversModuleNgSummary];
}
export function MdTabGroupNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdTabGroup,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ChangeDetectorRef}}}],
      lifecycleHooks:[1,4,5,7]},isComponent:true,selector:'md-tab-group, mat-tab-group',
      exportAs:(null as any),inputs:{color:'color',disableRipple:'disableRipple',dynamicHeight:'dynamicHeight',
          _dynamicHeightDeprecated:'md-dynamic-height',selectedIndex:'selectedIndex',
          headerPosition:'headerPosition',backgroundColor:'backgroundColor'},outputs:{selectedIndexChange:'selectedIndexChange',
          focusChange:'focusChange',selectChange:'selectChange'},hostListeners:{},
      hostProperties:{'class.mat-tab-group-dynamic-height':'dynamicHeight','class.mat-tab-group-inverted-header':'headerPosition === "below"'},
      hostAttributes:{class:'mat-tab-group'},providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i5.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:[{selectors:[{identifier:{reference:i0.MdTab}}],
          first:false,descendants:false,propertyName:'_tabs',read:(null as any)}],
      viewQueries:[{selectors:[{value:'tabBodyWrapper'}],first:true,descendants:true,
          propertyName:'_tabBodyWrapper',read:(null as any)}],entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:([] as any[]),
          encapsulation:2},componentViewType:i17.View_MdTabGroup_0,rendererType:i17.RenderType_MdTabGroup,
      componentFactory:i17.MdTabGroupNgFactory}];
}
export function MdTabLabelNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdTabLabel,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ViewContainerRef}}}],lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[md-tab-label], [mat-tab-label], [mdTabLabel], [matTabLabel]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MdTabNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdTab,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ViewContainerRef}}}],
      lifecycleHooks:[0,1,3]},isComponent:true,selector:'md-tab, mat-tab',exportAs:'mdTab, matTab',
      inputs:{disabled:'disabled',textLabel:'label'},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.MdTabLabel}}],
          first:true,descendants:true,propertyName:'templateLabel',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i2.TemplateRef}}],first:true,
          descendants:true,propertyName:'_content',read:(null as any)}],entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:['*'],
          encapsulation:2},componentViewType:i17.View_MdTab_0,rendererType:i17.RenderType_MdTab,
      componentFactory:i17.MdTabNgFactory}];
}
export function MdInkBarNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdInkBar,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'md-ink-bar, mat-ink-bar',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-ink-bar'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdTabLabelWrapperNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdTabLabelWrapper,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[mdTabLabelWrapper], [matTabLabelWrapper]',
      exportAs:(null as any),inputs:{disabled:'disabled'},outputs:{},hostListeners:{},
      hostProperties:{'class.mat-tab-disabled':'disabled'},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdTabNavNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdTabNav,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.Directionality}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ChangeDetectorRef}}}],
      lifecycleHooks:[1,4,5]},isComponent:true,selector:'[md-tab-nav-bar], [mat-tab-nav-bar]',
      exportAs:(null as any),inputs:{color:'color',disableRipple:'disableRipple',backgroundColor:'backgroundColor'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-tab-nav-bar'},
      providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i5.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:[{selectors:[{identifier:{reference:i0.MdTabLink}}],
          first:false,descendants:true,propertyName:'_tabLinks',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i0.MdInkBar}}],first:true,descendants:true,
          propertyName:'_inkBar',read:(null as any)}],entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:['*'],
          encapsulation:2},componentViewType:i17.View_MdTabNav_0,rendererType:i17.RenderType_MdTabNav,
      componentFactory:i17.MdTabNavNgFactory}];
}
export function MdTabLinkNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdTabLink,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.MdTabNav}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i7.ViewportRuler}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i5.MD_RIPPLE_GLOBAL_OPTIONS}}}],lifecycleHooks:[1]},
      isComponent:false,selector:'[md-tab-link], [mat-tab-link], [mdTabLink], [matTabLink]',
      exportAs:(null as any),inputs:{disabled:'disabled',active:'active'},outputs:{},
      hostListeners:{},hostProperties:{'attr.aria-disabled':'disabled.toString()',
          'attr.tabindex':'tabIndex','class.mat-tab-disabled':'disabled'},hostAttributes:{class:'mat-tab-link'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdTabBodyNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdTabBody,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i3.Directionality}}}],lifecycleHooks:[0,7]},
      isComponent:true,selector:'md-tab-body, mat-tab-body',exportAs:(null as any),
      inputs:{_content:'content',position:'position',origin:'origin'},outputs:{onCentering:'onCentering',
          onCentered:'onCentered'},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-tab-body'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:[{selectors:[{identifier:{reference:i9.PortalHostDirective}}],
          first:true,descendants:true,propertyName:'_portalHost',read:(null as any)}],
      entryComponents:([] as any[]),changeDetection:0,template:{animations:['translateTab'],
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i17.View_MdTabBody_0,
      rendererType:i17.RenderType_MdTabBody,componentFactory:i17.MdTabBodyNgFactory}];
}
export function MdTabHeaderNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdTabHeader,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.Renderer2}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ChangeDetectorRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i3.Directionality}}}],lifecycleHooks:[1,4,5]},
      isComponent:true,selector:'md-tab-header, mat-tab-header',exportAs:(null as any),
      inputs:{disableRipple:'disableRipple',selectedIndex:'selectedIndex'},outputs:{selectFocusedIndex:'selectFocusedIndex',
          indexFocused:'indexFocused'},hostListeners:{},hostProperties:{'class.mat-tab-header-pagination-controls-enabled':'_showPaginationControls',
          'class.mat-tab-header-rtl':'_getLayoutDirection() == \'rtl\''},hostAttributes:{class:'mat-tab-header'},
      providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i5.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:[{selectors:[{identifier:{reference:i0.MdTabLabelWrapper}}],
          first:false,descendants:false,propertyName:'_labelWrappers',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i0.MdInkBar}}],first:true,descendants:true,
          propertyName:'_inkBar',read:(null as any)},{selectors:[{value:'tabListContainer'}],
          first:true,descendants:true,propertyName:'_tabListContainer',read:(null as any)},
          {selectors:[{value:'tabList'}],first:true,descendants:true,propertyName:'_tabList',
              read:(null as any)}],entryComponents:([] as any[]),changeDetection:0,
      template:{animations:([] as any[]),ngContentSelectors:['*'],encapsulation:2},
      componentViewType:i17.View_MdTabHeader_0,rendererType:i17.RenderType_MdTabHeader,
      componentFactory:i17.MdTabHeaderNgFactory}];
}
