// @angular/cdk/overlay/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/cdk/overlay';
import * as i1 from '@angular/cdk/platform';
import * as i2 from '@angular/cdk/scrolling';
import * as i3 from '@angular/core';
import * as i4 from '@angular/cdk/portal';
import * as i5 from '../../portal/typings/index.ngsummary';
import * as i6 from '../../platform/typings/index.ngsummary';
import * as i7 from '../../scrolling/typings/index.ngsummary';
import * as i8 from '@angular/cdk/bidi';
export function OverlayModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.Platform,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i1.Platform,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
          useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
          multi:false},module:{reference:i1.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i2.ScrollDispatcher}},useClass:(null as any),
              useValue:(undefined as any),useFactory:{reference:i2.SCROLL_DISPATCHER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i2.ScrollDispatcher}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i3.NgZone}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i1.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i2.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i1.Platform}}}],multi:false},module:{reference:i2.ScrollDispatchModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.Overlay,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i0.ScrollStrategyOptions}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i0.OverlayContainer}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.ComponentFactoryResolver}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i0.ɵf}}},{isAttribute:false,isHost:false,
                      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.ApplicationRef}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Injector}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.NgZone}}}],lifecycleHooks:([] as any[])}},
              useClass:{reference:i0.Overlay,diDeps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ScrollStrategyOptions}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i0.OverlayContainer}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.ComponentFactoryResolver}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i0.ɵf}}},{isAttribute:false,isHost:false,
                      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.ApplicationRef}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Injector}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.NgZone}}}],lifecycleHooks:([] as any[])},
              useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
              deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i0.ScrollStrategyOptions}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i0.OverlayContainer}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.ComponentFactoryResolver}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i0.ɵf}}},{isAttribute:false,isHost:false,
                      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.ApplicationRef}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Injector}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.NgZone}}}],multi:false},module:{reference:i0.OverlayModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ɵf,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i2.ViewportRuler}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i0.ɵf,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ViewportRuler}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ViewportRuler}}}],
              multi:false},module:{reference:i0.OverlayModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i2.ViewportRuler}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i2.VIEWPORT_RULER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i2.ViewportRuler}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i2.ScrollDispatcher}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i2.ViewportRuler}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.ScrollDispatcher}}}],multi:false},
              module:{reference:i0.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i0.OverlayContainer}},useClass:(null as any),
              useValue:(undefined as any),useFactory:{reference:i0.ɵa,diDeps:[{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i0.OverlayContainer}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i0.OverlayContainer}}}],
              multi:false},module:{reference:i0.OverlayModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ɵc}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i0.ɵd,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                      isOptional:false,token:{identifier:{reference:i0.Overlay}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.Overlay}}}],
              multi:false},module:{reference:i0.OverlayModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ScrollStrategyOptions,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i2.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.ViewportRuler}}}],lifecycleHooks:([] as any[])}},
              useClass:{reference:i0.ScrollStrategyOptions,diDeps:[{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.ViewportRuler}}}],lifecycleHooks:([] as any[])},
              useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
              deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i2.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.ViewportRuler}}}],multi:false},
              module:{reference:i0.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],
      modules:[{reference:i4.PortalModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i1.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i2.ScrollDispatchModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i2.Scrollable},{reference:i0.ConnectedOverlayDirective},
          {reference:i0.OverlayOrigin}],exportedPipes:([] as any[])},ConnectedOverlayDirectiveNgSummary,
      OverlayOriginNgSummary,i5.PortalModuleNgSummary,i6.PlatformModuleNgSummary,i7.ScrollDispatchModuleNgSummary,
      {summaryKind:3,type:{reference:i0.Overlay,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ScrollStrategyOptions}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.OverlayContainer}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.ComponentFactoryResolver}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ɵf}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.ApplicationRef}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i3.Injector}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.NgZone}}}],
          lifecycleHooks:([] as any[])}},{summaryKind:3,type:{reference:i0.ɵf,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ViewportRuler}}}],
          lifecycleHooks:([] as any[])}},{summaryKind:3,type:{reference:i0.ScrollStrategyOptions,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i2.ScrollDispatcher}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ViewportRuler}}}],
          lifecycleHooks:([] as any[])}}];
}
export function ConnectedOverlayDirectiveNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.ConnectedOverlayDirective,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.Overlay}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i3.Renderer2}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.TemplateRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i3.ViewContainerRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ɵc}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i8.Directionality}}}],lifecycleHooks:[1,3]},
      isComponent:false,selector:'[cdk-connected-overlay], [connected-overlay], [cdkConnectedOverlay]',
      exportAs:'cdkConnectedOverlay',inputs:{origin:'cdkConnectedOverlayOrigin',positions:'cdkConnectedOverlayPositions',
          offsetX:'cdkConnectedOverlayOffsetX',offsetY:'cdkConnectedOverlayOffsetY',
          width:'cdkConnectedOverlayWidth',height:'cdkConnectedOverlayHeight',minWidth:'cdkConnectedOverlayMinWidth',
          minHeight:'cdkConnectedOverlayMinHeight',backdropClass:'cdkConnectedOverlayBackdropClass',
          scrollStrategy:'cdkConnectedOverlayScrollStrategy',open:'cdkConnectedOverlayOpen',
          hasBackdrop:'cdkConnectedOverlayHasBackdrop',_deprecatedOrigin:'origin',
          _deprecatedPositions:'positions',_deprecatedOffsetX:'offsetX',_deprecatedOffsetY:'offsetY',
          _deprecatedWidth:'width',_deprecatedHeight:'height',_deprecatedMinWidth:'minWidth',
          _deprecatedMinHeight:'minHeight',_deprecatedBackdropClass:'backdropClass',
          _deprecatedScrollStrategy:'scrollStrategy',_deprecatedOpen:'open',_deprecatedHasBackdrop:'hasBackdrop'},
      outputs:{backdropClick:'backdropClick',positionChange:'positionChange',attach:'attach',
          detach:'detach'},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function OverlayOriginNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.OverlayOrigin,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.ElementRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[cdk-overlay-origin], [overlay-origin], [cdkOverlayOrigin]',
      exportAs:'cdkOverlayOrigin',inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function OverlayNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.Overlay,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ScrollStrategyOptions}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.OverlayContainer}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.ComponentFactoryResolver}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ɵf}}},{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.ApplicationRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i3.Injector}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.NgZone}}}],
      lifecycleHooks:([] as any[])}}];
}
export function OverlayContainerNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.OverlayContainer,diDeps:([] as any[]),
      lifecycleHooks:[1]}}];
}
export function FullscreenOverlayContainerNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.FullscreenOverlayContainer,diDeps:([] as any[]),
      lifecycleHooks:[1]}}];
}
export function ScrollStrategyOptionsNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ScrollStrategyOptions,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ScrollDispatcher}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ViewportRuler}}}],lifecycleHooks:([] as any[])}}];
}
export function ɵfNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ɵf,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ViewportRuler}}}],
      lifecycleHooks:([] as any[])}}];
}
