// @angular/material/list/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/list';
import * as i1 from '@angular/cdk/bidi';
import * as i2 from '@angular/platform-browser';
import * as i3 from '@angular/material/core';
import * as i4 from '@angular/cdk/platform';
import * as i5 from '@angular/cdk/scrolling';
import * as i6 from '@angular/core';
import * as i7 from '@angular/common';
import * as i8 from '../../core/typings/index.ngsummary';
import * as i9 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i10 from '../../../cdk/platform/typings/index.ngsummary';
import * as i11 from '../../../cdk/scrolling/typings/index.ngsummary';
import * as i12 from '../../../common/common.ngsummary';
import * as i13 from './index.ngfactory';
export function MdListModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdListModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.DIR_DOCUMENT}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i2.DOCUMENT}},deps:(undefined as any),
          multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i1.Directionality,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i1.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i3.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i3.MdCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i2.DOCUMENT}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i4.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i4.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
              useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
              multi:false},module:{reference:i4.PlatformModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i5.ScrollDispatcher}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i5.SCROLL_DISPATCHER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i5.ScrollDispatcher}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.NgZone}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i4.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i5.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i4.Platform}}}],multi:false},module:{reference:i5.ScrollDispatchModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i5.ViewportRuler}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i5.VIEWPORT_RULER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i5.ViewportRuler}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i5.ScrollDispatcher}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i5.ViewportRuler}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.ScrollDispatcher}}}],multi:false},
              module:{reference:i3.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i7.NgLocalization}},useClass:{reference:i7.NgLocaleLocalization,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i6.LOCALE_ID}}}],lifecycleHooks:([] as any[])},
              useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
              deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i6.LOCALE_ID}}}],multi:false},
              module:{reference:i7.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],
      modules:[{reference:i3.CompatibilityModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i3.MdCommonModule,diDeps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:true,token:{identifier:{reference:i2.DOCUMENT}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                  token:{identifier:{reference:i3.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
          {reference:i3.MdLineModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i4.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i5.ScrollDispatchModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i3.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i3.MdPseudoCheckboxModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i7.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MdListModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i3.MatPrefixRejector},{reference:i3.MdPrefixRejector},
          {reference:i1.Dir},{reference:i3.MdLine},{reference:i3.MdPseudoCheckbox},
          {reference:i0.MdList},{reference:i0.MdListItem},{reference:i0.MdListDivider},
          {reference:i0.MdListAvatarCssMatStyler},{reference:i0.MdListIconCssMatStyler},
          {reference:i0.MdListCssMatStyler},{reference:i0.MdNavListCssMatStyler},{reference:i0.MdDividerCssMatStyler},
          {reference:i0.MdListSubheaderCssMatStyler},{reference:i0.MdSelectionList},
          {reference:i0.MdListOption}],exportedPipes:([] as any[])},MdListNgSummary,
      MdListItemNgSummary,MdListDividerNgSummary,MdListAvatarCssMatStylerNgSummary,
      MdListIconCssMatStylerNgSummary,MdListCssMatStylerNgSummary,MdNavListCssMatStylerNgSummary,
      MdDividerCssMatStylerNgSummary,MdListSubheaderCssMatStylerNgSummary,MdSelectionListNgSummary,
      MdListOptionNgSummary,i8.CompatibilityModuleNgSummary,i9.BidiModuleNgSummary,
      i8.MdCommonModuleNgSummary,i8.MdLineModuleNgSummary,i10.PlatformModuleNgSummary,
      i11.ScrollDispatchModuleNgSummary,i8.MdRippleModuleNgSummary,i8.MdPseudoCheckboxModuleNgSummary,
      i12.CommonModuleNgSummary];
}
export function MdListNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdList,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'md-list, mat-list, md-nav-list, mat-nav-list',exportAs:(null as any),
      inputs:{disableRipple:'disableRipple'},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{role:'list'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:['*'],
          encapsulation:2},componentViewType:i13.View_MdList_0,rendererType:i13.RenderType_MdList,
      componentFactory:i13.MdListNgFactory}];
}
export function MdListItemNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdListItem,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i6.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MdList}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MdNavListCssMatStyler}}}],lifecycleHooks:[4]},
      isComponent:true,selector:'md-list-item, mat-list-item, a[md-list-item], a[mat-list-item]',
      exportAs:(null as any),inputs:{disableRipple:'disableRipple'},outputs:{},hostListeners:{focus:'_handleFocus()',
          blur:'_handleBlur()'},hostProperties:{},hostAttributes:{role:'listitem',
          class:'mat-list-item'},providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i3.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:[{selectors:[{identifier:{reference:i3.MdLine}}],
          first:false,descendants:false,propertyName:'_lines',read:(null as any)},
          {selectors:[{identifier:{reference:i0.MdListAvatarCssMatStyler}}],first:true,
              descendants:true,propertyName:'_hasAvatar',read:(null as any)}],viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['[md-list-avatar], [md-list-icon], [mat-list-avatar], [mat-list-icon],\n              [mdListAvatar], [mdListIcon], [matListAvatar], [matListIcon]',
              '[md-line], [mat-line], [mdLine], [matLine]','*'],encapsulation:2},componentViewType:i13.View_MdListItem_0,
      rendererType:i13.RenderType_MdListItem,componentFactory:i13.MdListItemNgFactory}];
}
export function MdListDividerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdListDivider,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'md-divider, mat-divider',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{role:'separator',
          'aria-orientation':'horizontal'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdListAvatarCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdListAvatarCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[md-list-avatar], [mat-list-avatar], [mdListAvatar], [matListAvatar]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-list-avatar'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdListIconCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdListIconCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[md-list-icon], [mat-list-icon], [mdListIcon], [matListIcon]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-list-icon'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdListCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdListCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'md-list, mat-list',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-list'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdNavListCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdNavListCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'md-nav-list, mat-nav-list',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-nav-list'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdDividerCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdDividerCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'md-divider, mat-divider',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-divider'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdListSubheaderCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdListSubheaderCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[md-subheader], [mat-subheader], [mdSubheader], [matSubheader]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-subheader'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdSelectionListNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdSelectionList,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.ElementRef}}}],
      lifecycleHooks:[1,4]},isComponent:true,selector:'md-selection-list, mat-selection-list',
      exportAs:(null as any),inputs:{disabled:'disabled',disableRipple:'disableRipple'},
      outputs:{},hostListeners:{focus:'focus()',keydown:'_keydown($event)'},hostProperties:{'attr.tabindex':'_tabIndex',
          'attr.aria-disabled':'disabled.toString()'},hostAttributes:{role:'listbox',
          class:'mat-selection-list'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:[{selectors:[{identifier:{reference:i0.MdListOption}}],first:false,descendants:false,
          propertyName:'options',read:(null as any)}],viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:['*'],
          encapsulation:2},componentViewType:i13.View_MdSelectionList_0,rendererType:i13.RenderType_MdSelectionList,
      componentFactory:i13.MdSelectionListNgFactory}];
}
export function MdListOptionNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdListOption,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i6.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.ChangeDetectorRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MdSelectionList}}}],lifecycleHooks:[1,4]},
      isComponent:true,selector:'md-list-option, mat-list-option',exportAs:(null as any),
      inputs:{disableRipple:'disableRipple',checkboxPosition:'checkboxPosition',value:'value',
          disabled:'disabled',selected:'selected'},outputs:{selectChange:'selectChange',
          deselected:'deselected',destroyed:'destroyed'},hostListeners:{focus:'_handleFocus()',
          blur:'_handleBlur()',click:'_handleClick()'},hostProperties:{'class.mat-list-item-disabled':'disabled',
          'attr.aria-selected':'selected.toString()','attr.aria-disabled':'disabled.toString()'},
      hostAttributes:{role:'option',class:'mat-list-item mat-list-option',tabindex:'-1'},
      providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i3.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:[{selectors:[{identifier:{reference:i3.MdLine}}],
          first:false,descendants:false,propertyName:'_lines',read:(null as any)}],
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i13.View_MdListOption_0,
      rendererType:i13.RenderType_MdListOption,componentFactory:i13.MdListOptionNgFactory}];
}
