// @angular/material/expansion/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/expansion';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '@angular/cdk/platform';
import * as i4 from '@angular/cdk/a11y';
import * as i5 from '@angular/material/core';
import * as i6 from '../../core/typings/index.ngsummary';
import * as i7 from '../../../common/common.ngsummary';
import * as i8 from '../../../cdk/platform/typings/index.ngsummary';
import * as i9 from '../../../cdk/a11y/typings/index.ngsummary';
import * as i10 from './index.ngfactory';
export function MdExpansionModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdExpansionModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
      useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
      multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i3.Platform,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          useClass:{reference:i3.Platform,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:([] as any[]),multi:false},module:{reference:i3.PlatformModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i4.InteractivityChecker,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i3.Platform}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i4.InteractivityChecker,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          multi:false},module:{reference:i4.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i4.FocusTrapFactory,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.InteractivityChecker}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i3.Platform}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i4.FocusTrapFactory,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.InteractivityChecker}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i3.Platform}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i3.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
          multi:false},module:{reference:i4.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i4.AriaDescriber,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          lifecycleHooks:[1]}},useClass:{reference:i4.AriaDescriber,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          lifecycleHooks:[1]},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          multi:false},module:{reference:i4.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i4.LiveAnnouncer}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i4.LIVE_ANNOUNCER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i4.LiveAnnouncer}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i4.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i4.LiveAnnouncer}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i4.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i3.Platform}}}],multi:false},module:{reference:i4.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i4.AriaDescriber}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i4.ARIA_DESCRIBER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i4.AriaDescriber}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i4.AriaDescriber}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          multi:false},module:{reference:i4.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i4.FocusMonitor}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i4.FOCUS_MONITOR_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i4.FocusMonitor}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i4.FocusMonitor}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i3.Platform}}}],multi:false},module:{reference:i4.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i5.UniqueSelectionDispatcher}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i5.UNIQUE_SELECTION_DISPATCHER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i5.UniqueSelectionDispatcher}}}]},
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:true,isOptional:true,token:{identifier:{reference:i5.UniqueSelectionDispatcher}}}],
          multi:false},module:{reference:i0.MdExpansionModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}}],modules:[{reference:i5.CompatibilityModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i1.CommonModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i3.PlatformModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i4.A11yModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.MdExpansionModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i0.CdkAccordion},
      {reference:i0.MdAccordion},{reference:i0.MdExpansionPanel},{reference:i0.MdExpansionPanelActionRow},
      {reference:i0.MdExpansionPanelHeader},{reference:i0.MdExpansionPanelTitle},{reference:i0.MdExpansionPanelDescription}],
      exportedPipes:([] as any[])},CdkAccordionNgSummary,MdAccordionNgSummary,MdExpansionPanelNgSummary,
      MdExpansionPanelActionRowNgSummary,MdExpansionPanelHeaderNgSummary,MdExpansionPanelTitleNgSummary,
      MdExpansionPanelDescriptionNgSummary,i6.CompatibilityModuleNgSummary,i7.CommonModuleNgSummary,
      i8.PlatformModuleNgSummary,i9.A11yModuleNgSummary];
}
export function CdkAccordionNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkAccordion,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'cdk-accordion, [cdk-accordion]',exportAs:(null as any),
      inputs:{multi:'multi',hideToggle:'hideToggle',displayMode:'displayMode'},outputs:{},
      hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdAccordionNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdAccordion,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'mat-accordion, md-accordion',exportAs:(null as any),
      inputs:{multi:'multi',hideToggle:'hideToggle',displayMode:'displayMode'},outputs:{},
      hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-accordion'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdExpansionPanelNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdExpansionPanel,diDeps:[{isAttribute:false,
      isHost:true,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MdAccordion}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.UniqueSelectionDispatcher}}}],
      lifecycleHooks:[1,3]},isComponent:true,selector:'md-expansion-panel, mat-expansion-panel',
      exportAs:(null as any),inputs:{disabled:'disabled',expanded:'expanded',hideToggle:'hideToggle'},
      outputs:{},hostListeners:{},hostProperties:{'class.mat-expanded':'expanded',
          'class.mat-expansion-panel-spacing':'_hasSpacing()'},hostAttributes:{class:'mat-expansion-panel'},
      providers:[{token:{identifier:{reference:i0.AccordionItem}},useClass:(null as any),
          useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i0.MdExpansionPanel}},
          deps:(undefined as any),multi:false}],viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:['bodyExpansion'],
          ngContentSelectors:['mat-expansion-panel-header, md-expansion-panel-header',
              '*','mat-action-row, md-action-row'],encapsulation:2},componentViewType:i10.View_MdExpansionPanel_0,
      rendererType:i10.RenderType_MdExpansionPanel,componentFactory:i10.MdExpansionPanelNgFactory}];
}
export function MdExpansionPanelActionRowNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdExpansionPanelActionRow,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'mat-action-row, md-action-row',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-action-row'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdExpansionPanelHeaderNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdExpansionPanelHeader,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}},
      {isAttribute:false,isHost:true,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.MdExpansionPanel}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i4.FocusMonitor}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ChangeDetectorRef}}}],
      lifecycleHooks:[1]},isComponent:true,selector:'md-expansion-panel-header, mat-expansion-panel-header',
      exportAs:(null as any),inputs:{expandedHeight:'expandedHeight',collapsedHeight:'collapsedHeight'},
      outputs:{},hostListeners:{click:'_toggle()',keyup:'_keyup($event)'},hostProperties:{'attr.tabindex':'panel.disabled ? -1 : 0',
          'attr.aria-controls':'_getPanelId()','attr.aria-expanded':'_isExpanded()',
          'attr.aria-disabled':'panel.disabled','class.mat-expanded':'_isExpanded()',
          '@expansionHeight':'{\n        value: _getExpandedState(),\n        params: {\n          collapsedHeight: collapsedHeight,\n          expandedHeight: expandedHeight\n        }\n    }'},
      hostAttributes:{class:'mat-expansion-panel-header',role:'button'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:['indicatorRotate',
          'expansionHeight'],ngContentSelectors:['md-panel-title, mat-panel-title',
          'md-panel-description, mat-panel-description','*'],encapsulation:2},componentViewType:i10.View_MdExpansionPanelHeader_0,
      rendererType:i10.RenderType_MdExpansionPanelHeader,componentFactory:i10.MdExpansionPanelHeaderNgFactory}];
}
export function MdExpansionPanelTitleNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdExpansionPanelTitle,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'md-panel-title, mat-panel-title',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-expansion-panel-header-title'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdExpansionPanelDescriptionNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdExpansionPanelDescription,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'md-panel-description, mat-panel-description',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-expansion-panel-header-description'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function AccordionItemNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.AccordionItem,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.CdkAccordion}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.UniqueSelectionDispatcher}}}],
      lifecycleHooks:[1]}}];
}
export function ɵa14NgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ɵa14,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.MdAccordion}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.UniqueSelectionDispatcher}}}],
      lifecycleHooks:[1]}}];
}
