// @angular/cdk/a11y/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/cdk/a11y';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '@angular/cdk/platform';
import * as i4 from '../../../common/common.ngsummary';
import * as i5 from '../../platform/typings/index.ngsummary';
export function A11yModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
          useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i3.Platform,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},useClass:{reference:i3.Platform,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:([] as any[]),multi:false},module:{reference:i3.PlatformModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.InteractivityChecker,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i3.Platform}}}],lifecycleHooks:([] as any[])}},
              useClass:{reference:i0.InteractivityChecker,diDeps:[{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
                  lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
              multi:false},module:{reference:i0.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i0.FocusTrapFactory,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i3.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i0.FocusTrapFactory,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i0.InteractivityChecker}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}}],lifecycleHooks:([] as any[])},
              useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
              deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i0.InteractivityChecker}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}}],multi:false},module:{reference:i0.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.AriaDescriber,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i3.Platform}}}],lifecycleHooks:[1]}},
              useClass:{reference:i0.AriaDescriber,diDeps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
                  lifecycleHooks:[1]},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
              multi:false},module:{reference:i0.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i0.LiveAnnouncer}},useClass:(null as any),
              useValue:(undefined as any),useFactory:{reference:i0.LIVE_ANNOUNCER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i0.LiveAnnouncer}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:true,token:{identifier:{reference:i0.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i3.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i0.LiveAnnouncer}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i0.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}}],multi:false},module:{reference:i0.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.AriaDescriber}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i0.ARIA_DESCRIBER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i0.AriaDescriber}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i3.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i0.AriaDescriber}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}}],multi:false},module:{reference:i0.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.FocusMonitor}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i0.FOCUS_MONITOR_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i0.FocusMonitor}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i2.NgZone}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i3.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i0.FocusMonitor}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}}],multi:false},module:{reference:i0.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i1.CommonModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i3.PlatformModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i0.FocusTrapDirective},
          {reference:i0.FocusTrapDeprecatedDirective},{reference:i0.CdkMonitorFocus}],
      exportedPipes:([] as any[])},FocusTrapDirectiveNgSummary,FocusTrapDeprecatedDirectiveNgSummary,
      CdkMonitorFocusNgSummary,i4.CommonModuleNgSummary,i5.PlatformModuleNgSummary,
      {summaryKind:3,type:{reference:i0.InteractivityChecker,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          lifecycleHooks:([] as any[])}},{summaryKind:3,type:{reference:i0.FocusTrapFactory,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.InteractivityChecker}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i2.NgZone}}}],lifecycleHooks:([] as any[])}},
      {summaryKind:3,type:{reference:i0.AriaDescriber,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          lifecycleHooks:[1]}}];
}
export function FocusTrapDirectiveNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.FocusTrapDirective,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.FocusTrapFactory}}}],lifecycleHooks:[1,4]},
      isComponent:false,selector:'[cdkTrapFocus]',exportAs:'cdkTrapFocus',inputs:{enabled:'cdkTrapFocus'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function FocusTrapDeprecatedDirectiveNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.FocusTrapDeprecatedDirective,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.FocusTrapFactory}}}],lifecycleHooks:[1,4]},
      isComponent:false,selector:'cdk-focus-trap',exportAs:(null as any),inputs:{disabled:'disabled'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function CdkMonitorFocusNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkMonitorFocus,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.FocusMonitor}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}}],
      lifecycleHooks:[1]},isComponent:false,selector:'[cdkMonitorElementFocus], [cdkMonitorSubtreeFocus]',
      exportAs:(null as any),inputs:{},outputs:{cdkFocusChange:'cdkFocusChange'},hostListeners:{},
      hostProperties:{},hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function AriaDescriberNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.AriaDescriber,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
      lifecycleHooks:[1]}}];
}
export function FocusTrapFactoryNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.FocusTrapFactory,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.InteractivityChecker}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i3.Platform}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
      lifecycleHooks:([] as any[])}}];
}
export function InteractivityCheckerNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.InteractivityChecker,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
      lifecycleHooks:([] as any[])}}];
}
export function LiveAnnouncerNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.LiveAnnouncer,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i3.Platform}}}],lifecycleHooks:[1]}}];
}
export function FocusMonitorNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.FocusMonitor,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i3.Platform}}}],lifecycleHooks:([] as any[])}}];
}
