// @angular/material/button/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/button';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '@angular/cdk/bidi';
import * as i4 from '@angular/platform-browser';
import * as i5 from '@angular/material/core';
import * as i6 from '@angular/cdk/platform';
import * as i7 from '@angular/cdk/scrolling';
import * as i8 from '@angular/cdk/a11y';
import * as i9 from '../../../common/common.ngsummary';
import * as i10 from '../../core/typings/index.ngsummary';
import * as i11 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i12 from '../../../cdk/platform/typings/index.ngsummary';
import * as i13 from '../../../cdk/scrolling/typings/index.ngsummary';
import * as i14 from '../../../cdk/a11y/typings/index.ngsummary';
import * as i15 from './index.ngfactory';
export function MdButtonModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdButtonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
          useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i3.DIR_DOCUMENT}},useClass:(null as any),
              useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i4.DOCUMENT}},
              deps:(undefined as any),multi:false},module:{reference:i3.BidiModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i3.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i3.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i3.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i5.MdCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i4.DOCUMENT}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i6.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i6.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
              useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
              multi:false},module:{reference:i6.PlatformModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.ScrollDispatcher}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i7.SCROLL_DISPATCHER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i7.ScrollDispatcher}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i2.NgZone}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i7.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i7.ScrollDispatchModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.ViewportRuler}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i7.VIEWPORT_RULER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i7.ViewportRuler}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i7.ScrollDispatcher}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i7.ViewportRuler}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i7.ScrollDispatcher}}}],multi:false},
              module:{reference:i5.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i8.InteractivityChecker,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i8.InteractivityChecker,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i6.Platform}}}],lifecycleHooks:([] as any[])},
              useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
              deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i6.Platform}}}],multi:false},
              module:{reference:i8.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i8.FocusTrapFactory,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i8.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i6.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i8.FocusTrapFactory,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i8.InteractivityChecker}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}}],lifecycleHooks:([] as any[])},
              useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
              deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i8.InteractivityChecker}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}}],multi:false},module:{reference:i8.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i8.AriaDescriber,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i6.Platform}}}],lifecycleHooks:[1]}},
              useClass:{reference:i8.AriaDescriber,diDeps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
                  lifecycleHooks:[1]},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
              multi:false},module:{reference:i8.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i8.LiveAnnouncer}},useClass:(null as any),
              useValue:(undefined as any),useFactory:{reference:i8.LIVE_ANNOUNCER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i8.LiveAnnouncer}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:true,token:{identifier:{reference:i8.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i8.LiveAnnouncer}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i8.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i8.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i8.AriaDescriber}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i8.ARIA_DESCRIBER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i8.AriaDescriber}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i8.AriaDescriber}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i8.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i8.FocusMonitor}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i8.FOCUS_MONITOR_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i8.FocusMonitor}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i2.NgZone}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i8.FocusMonitor}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i8.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i1.CommonModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i5.CompatibilityModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i3.BidiModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i5.MdCommonModule,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i4.DOCUMENT}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],
          lifecycleHooks:([] as any[])},{reference:i6.PlatformModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])},{reference:i7.ScrollDispatchModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])},{reference:i5.MdRippleModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])},{reference:i8.A11yModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])},{reference:i0.MdButtonModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i5.MatPrefixRejector},
          {reference:i5.MdPrefixRejector},{reference:i3.Dir},{reference:i0.MdButton},
          {reference:i0.MdAnchor},{reference:i0.MdMiniFab},{reference:i0.MdFab},{reference:i0.MdButtonCssMatStyler},
          {reference:i0.MdRaisedButtonCssMatStyler},{reference:i0.MdIconButtonCssMatStyler}],
      exportedPipes:([] as any[])},MdButtonNgSummary,MdAnchorNgSummary,MdMiniFabNgSummary,
      MdFabNgSummary,MdButtonCssMatStylerNgSummary,MdRaisedButtonCssMatStylerNgSummary,
      MdIconButtonCssMatStylerNgSummary,i9.CommonModuleNgSummary,i10.CompatibilityModuleNgSummary,
      i11.BidiModuleNgSummary,i10.MdCommonModuleNgSummary,i12.PlatformModuleNgSummary,
      i13.ScrollDispatchModuleNgSummary,i10.MdRippleModuleNgSummary,i14.A11yModuleNgSummary];
}
export function MdButtonNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdButton,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i8.FocusMonitor}}}],lifecycleHooks:[1]},isComponent:true,
      selector:'button[md-button], button[md-raised-button], button[md-icon-button],\n             button[md-fab], button[md-mini-fab],\n             button[mat-button], button[mat-raised-button], button[mat-icon-button],\n             button[mat-fab], button[mat-mini-fab]',
      exportAs:(null as any),inputs:{disabled:'disabled',disableRipple:'disableRipple',
          color:'color'},outputs:{},hostListeners:{},hostProperties:{disabled:'disabled || null'},
      hostAttributes:{},providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i5.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i15.View_MdButton_0,
      rendererType:i15.RenderType_MdButton,componentFactory:i15.MdButtonNgFactory}];
}
export function MdAnchorNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdAnchor,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i8.FocusMonitor}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.Renderer2}}}],lifecycleHooks:[1]},isComponent:true,
      selector:'a[md-button], a[md-raised-button], a[md-icon-button], a[md-fab], a[md-mini-fab],\n             a[mat-button], a[mat-raised-button], a[mat-icon-button], a[mat-fab], a[mat-mini-fab]',
      exportAs:(null as any),inputs:{disabled:'disabled',disableRipple:'disableRipple',
          color:'color'},outputs:{},hostListeners:{click:'_haltDisabledEvents($event)'},
      hostProperties:{'attr.tabindex':'disabled ? -1 : 0','attr.disabled':'disabled || null',
          'attr.aria-disabled':'disabled.toString()'},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i15.View_MdAnchor_0,
      rendererType:i15.RenderType_MdAnchor,componentFactory:i15.MdAnchorNgFactory}];
}
export function MdMiniFabNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdMiniFab,diDeps:[{isAttribute:false,isHost:false,
      isSelf:true,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MdButton}}},
      {isAttribute:false,isHost:false,isSelf:true,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MdAnchor}}}],lifecycleHooks:([] as any[])},
      isComponent:false,selector:'button[md-mini-fab], button[mat-mini-fab], a[md-mini-fab], a[mat-mini-fab]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-mini-fab'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdFabNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdFab,diDeps:[{isAttribute:false,isHost:false,
      isSelf:true,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MdButton}}},
      {isAttribute:false,isHost:false,isSelf:true,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MdAnchor}}}],lifecycleHooks:([] as any[])},
      isComponent:false,selector:'button[md-fab], button[mat-fab], a[md-fab], a[mat-fab]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-fab'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdButtonCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdButtonCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'button[md-button], button[mat-button], a[md-button], a[mat-button]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-button'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdRaisedButtonCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdRaisedButtonCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'button[md-raised-button], button[mat-raised-button], a[md-raised-button], a[mat-raised-button]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-raised-button'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdIconButtonCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdIconButtonCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'button[md-icon-button], button[mat-icon-button], a[md-icon-button], a[mat-icon-button]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-icon-button'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
