// @angular/cdk/stepper/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/cdk/stepper';
import * as i1 from '@angular/cdk/bidi';
import * as i2 from '@angular/platform-browser';
import * as i3 from '@angular/common';
import * as i4 from '@angular/core';
import * as i5 from '../../bidi/typings/index.ngsummary';
import * as i6 from '../../../common/common.ngsummary';
import * as i7 from './index.ngfactory';
export function CdkStepperModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.CdkStepperModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.DIR_DOCUMENT}},
      useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
      useExisting:{identifier:{reference:i2.DOCUMENT}},deps:(undefined as any),multi:false},
      module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i1.Directionality,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i1.Directionality,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
          multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i3.NgLocalization}},useClass:{reference:i3.NgLocaleLocalization,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i4.LOCALE_ID}}}],lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i4.LOCALE_ID}}}],multi:false},module:{reference:i3.CommonModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i1.BidiModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i3.CommonModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.CdkStepperModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i0.CdkStep},
      {reference:i0.CdkStepper},{reference:i0.CdkStepLabel},{reference:i0.CdkStepperNext},
      {reference:i0.CdkStepperPrevious}],exportedPipes:([] as any[])},CdkStepNgSummary,
      CdkStepperNgSummary,CdkStepLabelNgSummary,CdkStepperNextNgSummary,CdkStepperPreviousNgSummary,
      i5.BidiModuleNgSummary,i6.CommonModuleNgSummary];
}
export function CdkStepNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkStep,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.CdkStepper}}}],
      lifecycleHooks:([] as any[])},isComponent:true,selector:'cdk-step',exportAs:(null as any),
      inputs:{stepControl:'stepControl',label:'label',editable:'editable',optional:'optional',
          completed:'completed'},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},
      providers:([] as any[]),viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.CdkStepLabel}}],
          first:true,descendants:true,propertyName:'stepLabel',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i4.TemplateRef}}],first:true,
          descendants:true,propertyName:'content',read:(null as any)}],entryComponents:([] as any[]),
      changeDetection:1,template:{animations:([] as any[]),ngContentSelectors:['*'],
          encapsulation:2},componentViewType:i7.View_CdkStep_0,rendererType:i7.RenderType_CdkStep,
      componentFactory:i7.CdkStepNgFactory}];
}
export function CdkStepperNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkStepper,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.Directionality}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[cdkStepper]',exportAs:(null as any),
      inputs:{linear:'linear',selectedIndex:'selectedIndex',selected:'selected'},outputs:{selectionChange:'selectionChange'},
      hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.CdkStep}}],
          first:false,descendants:false,propertyName:'_steps',read:(null as any)}],
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function CdkStepLabelNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkStepLabel,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.TemplateRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[cdkStepLabel]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function CdkStepperNextNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkStepperNext,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.CdkStepper}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'button[cdkStepperNext]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{click:'_stepper.next()'},
      hostProperties:{},hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function CdkStepperPreviousNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkStepperPrevious,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.CdkStepper}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'button[cdkStepperPrevious]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{click:'_stepper.previous()'},
      hostProperties:{},hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
