// @angular/material/table/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/table';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '@angular/cdk/bidi';
import * as i4 from '@angular/platform-browser';
import * as i5 from '@angular/material/core';
import * as i6 from '@angular/cdk/table';
import * as i7 from '../../../common/common.ngsummary';
import * as i8 from '../../../cdk/table/typings/index.ngsummary';
import * as i9 from '../../core/typings/index.ngsummary';
import * as i10 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i11 from './index.ngfactory';
export function MdTableModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdTableModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
          useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i3.DIR_DOCUMENT}},useClass:(null as any),
              useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i4.DOCUMENT}},
              deps:(undefined as any),multi:false},module:{reference:i3.BidiModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i3.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i3.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i3.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i5.MdCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i4.DOCUMENT}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}}],modules:[{reference:i1.CommonModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i6.CdkTableModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i5.CompatibilityModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i3.BidiModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i5.MdCommonModule,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i4.DOCUMENT}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],
          lifecycleHooks:([] as any[])},{reference:i0.MdTableModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i0.MdTable},
          {reference:i0.MdCellDef},{reference:i0.MdHeaderCellDef},{reference:i0.MdColumnDef},
          {reference:i0.MdHeaderRowDef},{reference:i0.MdRowDef},{reference:i0.MdHeaderCell},
          {reference:i0.MdCell},{reference:i0.MdHeaderRow},{reference:i0.MdRow},{reference:i0.MatHeaderRowDef},
          {reference:i0.MatRowDef}],exportedPipes:([] as any[])},MdTableNgSummary,
      MdCellDefNgSummary,MdHeaderCellDefNgSummary,MdColumnDefNgSummary,MdHeaderRowDefNgSummary,
      MdRowDefNgSummary,MdHeaderCellNgSummary,MdCellNgSummary,MdHeaderRowNgSummary,
      MdRowNgSummary,MatHeaderRowDefNgSummary,MatRowDefNgSummary,i7.CommonModuleNgSummary,
      i8.CdkTableModuleNgSummary,i9.CompatibilityModuleNgSummary,i10.BidiModuleNgSummary,
      i9.MdCommonModuleNgSummary];
}
export function MdTableNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdTable,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.IterableDiffers}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.Renderer2}}},{isAttribute:true,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{value:'role'}}],lifecycleHooks:[0,
      1,4,5]},isComponent:true,selector:'md-table, mat-table',exportAs:(null as any),
      inputs:{trackBy:'trackBy',dataSource:'dataSource'},outputs:{},hostListeners:{},
      hostProperties:{},hostAttributes:{class:'mat-table'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i6.CdkColumnDef}}],
          first:false,descendants:false,propertyName:'_columnDefinitions',read:(null as any)},
          {selectors:[{identifier:{reference:i6.CdkHeaderRowDef}}],first:true,descendants:true,
              propertyName:'_headerDefinition',read:(null as any)},{selectors:[{identifier:{reference:i6.CdkRowDef}}],
              first:false,descendants:false,propertyName:'_rowDefinitions',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i6.RowPlaceholder}}],first:true,
          descendants:true,propertyName:'_rowPlaceholder',read:(null as any)},{selectors:[{identifier:{reference:i6.HeaderRowPlaceholder}}],
          first:true,descendants:true,propertyName:'_headerRowPlaceholder',read:(null as any)}],
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i11.View_MdTable_0,
      rendererType:i11.RenderType_MdTable,componentFactory:i11.MdTableNgFactory}];
}
export function MdCellDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdCellDef,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[mdCellDef], [matCellDef]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkCellDef}},useClass:(null as any),
          useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i0.MdCellDef}},
          deps:(undefined as any),multi:false}],viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MdHeaderCellDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdHeaderCellDef,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[mdHeaderCellDef], [matHeaderCellDef]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkHeaderCellDef}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MdHeaderCellDef}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdColumnDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdColumnDef,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[mdColumnDef], [matColumnDef]',exportAs:(null as any),
      inputs:{name:'mdColumnDef',_matColumnDefName:'matColumnDef'},outputs:{},hostListeners:{},
      hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkColumnDef}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MdColumnDef}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i6.CdkCellDef}}],
          first:true,descendants:true,propertyName:'cell',read:(null as any)},{selectors:[{identifier:{reference:i6.CdkHeaderCellDef}}],
          first:true,descendants:true,propertyName:'headerCell',read:(null as any)}],
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MdHeaderRowDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdHeaderRowDef,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.IterableDiffers}}}],lifecycleHooks:[3]},
      isComponent:false,selector:'[mdHeaderRowDef]',exportAs:(null as any),inputs:{columns:'mdHeaderRowDef'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkHeaderRowDef}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MdHeaderRowDef}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdRowDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdRowDef,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.IterableDiffers}}}],lifecycleHooks:[3]},
      isComponent:false,selector:'[mdRowDef]',exportAs:(null as any),inputs:{columns:'mdRowDefColumns'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkRowDef}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MdRowDef}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdHeaderCellNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdHeaderCell,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.CdkColumnDef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'md-header-cell, mat-header-cell',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-header-cell',role:'columnheader'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdCellNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdCell,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.CdkColumnDef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'md-cell, mat-cell',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-cell',role:'gridcell'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdHeaderRowNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdHeaderRow,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'md-header-row, mat-header-row',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-header-row',
          role:'row'},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i11.View_MdHeaderRow_0,
      rendererType:i11.RenderType_MdHeaderRow,componentFactory:i11.MdHeaderRowNgFactory}];
}
export function MdRowNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdRow,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'md-row, mat-row',exportAs:(null as any),inputs:{},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-row',
          role:'row'},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i11.View_MdRow_0,
      rendererType:i11.RenderType_MdRow,componentFactory:i11.MdRowNgFactory}];
}
export function MatHeaderRowDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatHeaderRowDef,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.IterableDiffers}}}],lifecycleHooks:[3]},
      isComponent:false,selector:'[matHeaderRowDef]',exportAs:(null as any),inputs:{columns:'matHeaderRowDef'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkHeaderRowDef}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatHeaderRowDef}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatRowDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatRowDef,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.IterableDiffers}}}],lifecycleHooks:[3]},
      isComponent:false,selector:'[matRowDef]',exportAs:(null as any),inputs:{columns:'matRowDefColumns'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkRowDef}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatRowDef}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
