// @angular/material/datepicker/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/datepicker';
import * as i1 from '@angular/material/dialog';
import * as i2 from '../../dialog/typings/index.ngfactory';
import * as i3 from './index.ngfactory';
import * as i4 from '@angular/common';
import * as i5 from '@angular/core';
import * as i6 from '@angular/cdk/bidi';
import * as i7 from '@angular/platform-browser';
import * as i8 from '@angular/material/core';
import * as i9 from '@angular/cdk/platform';
import * as i10 from '@angular/cdk/scrolling';
import * as i11 from '@angular/cdk/a11y';
import * as i12 from '@angular/cdk/overlay';
import * as i13 from '@angular/material/icon';
import * as i14 from '@angular/http';
import * as i15 from '@angular/material/button';
import * as i16 from '@angular/cdk/portal';
import * as i17 from '../../../common/common.ngsummary';
import * as i18 from '../../core/typings/index.ngsummary';
import * as i19 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i20 from '../../../cdk/platform/typings/index.ngsummary';
import * as i21 from '../../../cdk/scrolling/typings/index.ngsummary';
import * as i22 from '../../../cdk/a11y/typings/index.ngsummary';
import * as i23 from '../../button/typings/index.ngsummary';
import * as i24 from '../../../cdk/portal/typings/index.ngsummary';
import * as i25 from '../../../cdk/overlay/typings/index.ngsummary';
import * as i26 from '../../dialog/typings/index.ngsummary';
import * as i27 from '../../icon/typings/index.ngsummary';
import * as i28 from '@angular/material/form-field';
import * as i29 from '@angular/forms';
export function MdDatepickerModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdDatepickerModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:[{componentType:i1.MdDialogContainer,
      componentFactory:i2.MdDialogContainerNgFactory},{componentType:i0.MdDatepickerContent,
      componentFactory:i3.MdDatepickerContentNgFactory}],providers:[{provider:{token:{identifier:{reference:i4.NgLocalization}},
      useClass:{reference:i4.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.LOCALE_ID}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.LOCALE_ID}}}],
      multi:false},module:{reference:i4.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i6.DIR_DOCUMENT}},useClass:(null as any),
          useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i7.DOCUMENT}},
          deps:(undefined as any),multi:false},module:{reference:i6.BidiModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i6.Directionality,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i6.DIR_DOCUMENT}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i6.Directionality,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i6.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:true,token:{identifier:{reference:i6.DIR_DOCUMENT}}}],
          multi:false},module:{reference:i6.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i8.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
          useValue:true,useFactory:(null as any),useExisting:(undefined as any),deps:(undefined as any),
          multi:false},module:{reference:i8.MdCommonModule,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i7.DOCUMENT}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i8.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i9.Platform,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          useClass:{reference:i9.Platform,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:([] as any[]),multi:false},module:{reference:i9.PlatformModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i10.ScrollDispatcher}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i10.SCROLL_DISPATCHER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i10.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i9.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i10.ScrollDispatcher}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i9.Platform}}}],multi:false},module:{reference:i10.ScrollDispatchModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i10.ViewportRuler}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i10.VIEWPORT_RULER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i10.ViewportRuler}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i10.ScrollDispatcher}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i10.ViewportRuler}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i10.ScrollDispatcher}}}],
          multi:false},module:{reference:i8.MdRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i11.InteractivityChecker,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.Platform}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i11.InteractivityChecker,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i9.Platform}}}],lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i9.Platform}}}],multi:false},module:{reference:i11.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i11.FocusTrapFactory,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i11.InteractivityChecker}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.Platform}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i5.NgZone}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i11.FocusTrapFactory,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i11.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i9.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i11.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i9.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}}],
          multi:false},module:{reference:i11.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i11.AriaDescriber,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.Platform}}}],
          lifecycleHooks:[1]}},useClass:{reference:i11.AriaDescriber,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.Platform}}}],
          lifecycleHooks:[1]},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.Platform}}}],
          multi:false},module:{reference:i11.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i11.LiveAnnouncer}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i11.LIVE_ANNOUNCER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i11.LiveAnnouncer}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i11.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i9.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i11.LiveAnnouncer}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i11.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i9.Platform}}}],multi:false},module:{reference:i11.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i11.AriaDescriber}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i11.ARIA_DESCRIBER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i11.AriaDescriber}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i9.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i11.AriaDescriber}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.Platform}}}],
          multi:false},module:{reference:i11.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i11.FocusMonitor}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i11.FOCUS_MONITOR_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i11.FocusMonitor}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i9.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i11.FocusMonitor}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i9.Platform}}}],multi:false},module:{reference:i11.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i12.Overlay,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i12.ScrollStrategyOptions}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.OverlayContainer}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i5.ComponentFactoryResolver}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.ɵf}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i5.ApplicationRef}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.Injector}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i5.NgZone}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i12.Overlay,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.ScrollStrategyOptions}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i12.OverlayContainer}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.ComponentFactoryResolver}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i12.ɵf}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.ApplicationRef}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i5.Injector}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.ScrollStrategyOptions}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i12.OverlayContainer}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.ComponentFactoryResolver}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i12.ɵf}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.ApplicationRef}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i5.Injector}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}}],
          multi:false},module:{reference:i12.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i12.ɵf,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i10.ViewportRuler}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i12.ɵf,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i10.ViewportRuler}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i10.ViewportRuler}}}],
          multi:false},module:{reference:i12.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i10.ViewportRuler}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i10.VIEWPORT_RULER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i10.ViewportRuler}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i10.ScrollDispatcher}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i10.ViewportRuler}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i10.ScrollDispatcher}}}],
          multi:false},module:{reference:i12.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i12.OverlayContainer}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i12.ɵa,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i12.OverlayContainer}}}]},
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:true,isOptional:true,token:{identifier:{reference:i12.OverlayContainer}}}],
          multi:false},module:{reference:i12.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i12.ɵc}},useClass:(null as any),useValue:(undefined as any),
          useFactory:{reference:i12.ɵd,diDeps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.Overlay}}}]},
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.Overlay}}}],
          multi:false},module:{reference:i12.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i12.ScrollStrategyOptions,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i10.ScrollDispatcher}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i10.ViewportRuler}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i12.ScrollStrategyOptions,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i10.ScrollDispatcher}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i10.ViewportRuler}}}],lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i10.ScrollDispatcher}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i10.ViewportRuler}}}],
          multi:false},module:{reference:i12.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i1.MdDialog,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.Overlay}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i5.Injector}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i4.Location}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i1.MD_DIALOG_SCROLL_STRATEGY}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i1.MdDialog}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i1.MdDialog,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.Overlay}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i5.Injector}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i4.Location}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i1.MD_DIALOG_SCROLL_STRATEGY}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i1.MdDialog}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.Overlay}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i5.Injector}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i4.Location}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i1.MD_DIALOG_SCROLL_STRATEGY}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i1.MdDialog}}}],
          multi:false},module:{reference:i1.MdDialogModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i1.MD_DIALOG_SCROLL_STRATEGY}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i1.MAT_DIALOG_SCROLL_STRATEGY_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i12.Overlay}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i12.Overlay}}}],multi:false},module:{reference:i1.MdDialogModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i13.MdIconRegistry}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i13.ICON_REGISTRY_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i13.MdIconRegistry}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i14.Http}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i7.DomSanitizer}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i13.MdIconRegistry}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i14.Http}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i7.DomSanitizer}}}],multi:false},module:{reference:i13.MdIconModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.MdDatepickerIntl,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i0.MdDatepickerIntl,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
          useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
          multi:false},module:{reference:i0.MdDatepickerModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.MD_DATEPICKER_SCROLL_STRATEGY}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i0.MAT_DATEPICKER_SCROLL_STRATEGY_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i12.Overlay}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i12.Overlay}}}],multi:false},module:{reference:i0.MdDatepickerModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i4.CommonModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i8.CompatibilityModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i6.BidiModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i8.MdCommonModule,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i7.DOCUMENT}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i8.MATERIAL_SANITY_CHECKS}}}],
      lifecycleHooks:([] as any[])},{reference:i9.PlatformModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i10.ScrollDispatchModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i8.MdRippleModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i11.A11yModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i15.MdButtonModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i16.PortalModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i12.OverlayModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i1.MdDialogModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i13.MdIconModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i0.MdDatepickerModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i0.MdCalendar},
      {reference:i0.MdCalendarBody},{reference:i0.MdDatepicker},{reference:i0.MdDatepickerContent},
      {reference:i0.MdDatepickerInput},{reference:i0.MdDatepickerToggle},{reference:i0.MdMonthView},
      {reference:i0.MdYearView}],exportedPipes:([] as any[])},MdCalendarNgSummary,
      MdCalendarBodyNgSummary,MdDatepickerNgSummary,MdDatepickerContentNgSummary,MdDatepickerInputNgSummary,
      MdDatepickerToggleNgSummary,MdMonthViewNgSummary,MdYearViewNgSummary,i17.CommonModuleNgSummary,
      i18.CompatibilityModuleNgSummary,i19.BidiModuleNgSummary,i18.MdCommonModuleNgSummary,
      i20.PlatformModuleNgSummary,i21.ScrollDispatchModuleNgSummary,i18.MdRippleModuleNgSummary,
      i22.A11yModuleNgSummary,i23.MdButtonModuleNgSummary,i24.PortalModuleNgSummary,
      i25.OverlayModuleNgSummary,i26.MdDialogModuleNgSummary,i27.MdIconModuleNgSummary,
      {summaryKind:3,type:{reference:i0.MdDatepickerIntl,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
export function MdCalendarNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdCalendar,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.MdDatepickerIntl}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i8.DateAdapter}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i8.MD_DATE_FORMATS}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i5.ChangeDetectorRef}}}],lifecycleHooks:[1,
      4]},isComponent:true,selector:'md-calendar, mat-calendar',exportAs:(null as any),
      inputs:{startAt:'startAt',startView:'startView',selected:'selected',minDate:'minDate',
          maxDate:'maxDate',dateFilter:'dateFilter'},outputs:{selectedChange:'selectedChange',
          userSelection:'userSelection'},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-calendar'},
      providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i8.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i3.View_MdCalendar_0,
      rendererType:i3.RenderType_MdCalendar,componentFactory:i3.MdCalendarNgFactory}];
}
export function MdCalendarBodyNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdCalendarBody,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'[md-calendar-body], [mat-calendar-body]',exportAs:(null as any),
      inputs:{label:'label',rows:'rows',todayValue:'todayValue',selectedValue:'selectedValue',
          labelMinRequiredCells:'labelMinRequiredCells',numCols:'numCols',allowDisabledSelection:'allowDisabledSelection',
          activeCell:'activeCell',cellAspectRatio:'cellAspectRatio'},outputs:{selectedValueChange:'selectedValueChange'},
      hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-calendar-body'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i3.View_MdCalendarBody_0,
      rendererType:i3.RenderType_MdCalendarBody,componentFactory:i3.MdCalendarBodyNgFactory}];
}
export function MdDatepickerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdDatepicker,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.MdDialog}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i12.Overlay}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i5.ViewContainerRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.MD_DATEPICKER_SCROLL_STRATEGY}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i8.DateAdapter}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i6.Directionality}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i7.DOCUMENT}}}],lifecycleHooks:[1]},isComponent:true,
      selector:'md-datepicker, mat-datepicker',exportAs:(null as any),inputs:{startAt:'startAt',
          startView:'startView',touchUi:'touchUi',disabled:'disabled'},outputs:{selectedChanged:'selectedChanged'},
      hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i3.View_MdDatepicker_0,
      rendererType:i3.RenderType_MdDatepicker,componentFactory:i3.MdDatepickerNgFactory}];
}
export function MdDatepickerContentNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdDatepickerContent,diDeps:([] as any[]),
      lifecycleHooks:[4]},isComponent:true,selector:'md-datepicker-content, mat-datepicker-content',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{keydown:'_handleKeydown($event)'},
      hostProperties:{'class.mat-datepicker-content-touch':'datepicker.touchUi'},hostAttributes:{class:'mat-datepicker-content'},
      providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i8.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:([] as any[]),viewQueries:[{selectors:[{identifier:{reference:i0.MdCalendar}}],
          first:true,descendants:true,propertyName:'_calendar',read:(null as any)}],
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i3.View_MdDatepickerContent_0,
      rendererType:i3.RenderType_MdDatepickerContent,componentFactory:i3.MdDatepickerContentNgFactory}];
}
export function MdDatepickerInputNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdDatepickerInput,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i5.Renderer2}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i8.DateAdapter}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i8.MD_DATE_FORMATS}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i28.MdFormField}}}],
      lifecycleHooks:[1,4]},isComponent:false,selector:'input[mdDatepicker], input[matDatepicker]',
      exportAs:'mdDatepickerInput, matDatepickerInput',inputs:{mdDatepicker:'mdDatepicker',
          matDatepicker:'matDatepicker',mdDatepickerFilter:'mdDatepickerFilter',matDatepickerFilter:'matDatepickerFilter',
          value:'value',min:'min',max:'max',disabled:'disabled'},outputs:{dateChange:'dateChange',
          dateInput:'dateInput'},hostListeners:{input:'_onInput($event.target.value)',
          change:'_onChange()',blur:'_onTouched()',keydown:'_onKeydown($event)'},hostProperties:{'attr.aria-haspopup':'true',
          'attr.aria-owns':'(_datepicker?.opened && _datepicker.id) || null','attr.min':'min ? _dateAdapter.toIso8601(min) : null',
          'attr.max':'max ? _dateAdapter.toIso8601(max) : null',disabled:'disabled'},
      hostAttributes:{},providers:[{token:{identifier:{reference:i29.NG_VALUE_ACCESSOR}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MdDatepickerInput}},deps:(undefined as any),
          multi:true},{token:{identifier:{reference:i29.NG_VALIDATORS}},useClass:(null as any),
          useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i0.MdDatepickerInput}},
          deps:(undefined as any),multi:true}],viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MdDatepickerToggleNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdDatepickerToggle,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.MdDatepickerIntl}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i5.ChangeDetectorRef}}}],lifecycleHooks:[1,
      3]},isComponent:true,selector:'md-datepicker-toggle, mat-datepicker-toggle',
      exportAs:(null as any),inputs:{datepicker:'for',disabled:'disabled'},outputs:{},
      hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-datepicker-toggle'},
      providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i8.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i3.View_MdDatepickerToggle_0,
      rendererType:i3.RenderType_MdDatepickerToggle,componentFactory:i3.MdDatepickerToggleNgFactory}];
}
export function MdMonthViewNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdMonthView,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i8.DateAdapter}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i8.MD_DATE_FORMATS}}}],lifecycleHooks:[4]},
      isComponent:true,selector:'md-month-view',exportAs:(null as any),inputs:{activeDate:'activeDate',
          selected:'selected',dateFilter:'dateFilter'},outputs:{selectedChange:'selectedChange',
          userSelection:'userSelection'},hostListeners:{},hostProperties:{},hostAttributes:{},
      providers:([] as any[]),viewProviders:[{token:{identifier:{reference:i8.MATERIAL_COMPATIBILITY_MODE}},
          useClass:(null as any),useValue:true,useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false}],queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i3.View_MdMonthView_0,
      rendererType:i3.RenderType_MdMonthView,componentFactory:i3.MdMonthViewNgFactory}];
}
export function MdYearViewNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdYearView,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i8.DateAdapter}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i8.MD_DATE_FORMATS}}}],lifecycleHooks:[4]},
      isComponent:true,selector:'md-year-view, mat-year-view',exportAs:(null as any),
      inputs:{activeDate:'activeDate',selected:'selected',dateFilter:'dateFilter'},
      outputs:{selectedChange:'selectedChange'},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i3.View_MdYearView_0,
      rendererType:i3.RenderType_MdYearView,componentFactory:i3.MdYearViewNgFactory}];
}
export function MdDatepickerIntlNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.MdDatepickerIntl,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}}];
}
