// @angular/material/chips/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/chips';
import * as i1 from '@angular/core';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '@angular/forms';
import * as i4 from '@angular/material/form-field';
import * as i5 from './index.ngfactory';
export function MdChipsModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MdChipsModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:([] as any[]),modules:[{reference:i0.MdChipsModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i0.MdChipList},
          {reference:i0.MdChip},{reference:i0.MdChipInput},{reference:i0.MdChipRemove},
          {reference:i0.MdBasicChip}],exportedPipes:([] as any[])},MdChipListNgSummary,
      MdChipNgSummary,MdChipInputNgSummary,MdChipRemoveNgSummary,MdBasicChipNgSummary];
}
export function MdChipListNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdChipList,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ChangeDetectorRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i2.Directionality}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.NgForm}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i3.FormGroupDirective}}},{isAttribute:false,
          isHost:false,isSelf:true,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.NgControl}}}],
      lifecycleHooks:[0,1,4]},isComponent:true,selector:'md-chip-list, mat-chip-list',
      exportAs:'mdChipList, matChipList',inputs:{multiple:'multiple',compareWith:'compareWith',
          value:'value',id:'id',required:'required',placeholder:'placeholder',disabled:'disabled',
          ariaOrientation:'aria-orientation',selectable:'selectable',tabIndex:'tabIndex'},
      outputs:{change:'change',valueChange:'valueChange'},hostListeners:{focus:'focus()',
          blur:'_blur()',keydown:'_keydown($event)'},hostProperties:{'attr.tabindex':'_tabIndex',
          'attr.aria-describedby':'_ariaDescribedby || null','attr.aria-required':'required.toString()',
          'attr.aria-disabled':'disabled.toString()','attr.aria-invalid':'errorState',
          'attr.aria-multiselectable':'multiple','class.mat-chip-list-disabled':'disabled',
          'class.mat-chip-list-invalid':'errorState','class.mat-chip-list-required':'required',
          'attr.aria-orientation':'ariaOrientation'},hostAttributes:{role:'listbox',
          class:'mat-chip-list'},providers:[{token:{identifier:{reference:i4.MdFormFieldControl}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MdChipList}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.MdChip}}],
          first:false,descendants:false,propertyName:'chips',read:(null as any)}],
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i5.View_MdChipList_0,
      rendererType:i5.RenderType_MdChipList,componentFactory:i5.MdChipListNgFactory}];
}
export function MdChipNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdChip,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.ElementRef}}}],lifecycleHooks:[1]},isComponent:false,
      selector:'md-basic-chip, [md-basic-chip], md-chip, [md-chip],\n             mat-basic-chip, [mat-basic-chip], mat-chip, [mat-chip]',
      exportAs:'mdChip, matChip',inputs:{color:'color',disabled:'disabled',selected:'selected',
          value:'value',selectable:'selectable',removable:'removable'},outputs:{selectionChange:'selectionChange',
          destroyed:'destroyed',destroy:'destroy',removed:'removed',onRemove:'remove'},
      hostListeners:{click:'_handleClick($event)',keydown:'_handleKeydown($event)',
          focus:'_hasFocus = true',blur:'_blur()'},hostProperties:{'class.mat-chip-selected':'selected',
          'attr.disabled':'disabled || null','attr.aria-disabled':'disabled.toString()',
          'attr.aria-selected':'ariaSelected'},hostAttributes:{class:'mat-chip',tabindex:'-1',
          role:'option'},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MdChipInputNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdChipInput,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.ElementRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'input[mdChipInputFor], input[matChipInputFor]',
      exportAs:(null as any),inputs:{chipList:'mdChipInputFor',addOnBlur:'mdChipInputAddOnBlur',
          separatorKeyCodes:'mdChipInputSeparatorKeyCodes',matChipList:'matChipInputFor',
          matAddOnBlur:'matChipInputAddOnBlur',matSeparatorKeyCodes:'matChipInputSeparatorKeyCodes',
          placeholder:'placeholder'},outputs:{chipEnd:'mdChipInputTokenEnd',_matChipInputTokenEnd:'matChipInputTokenEnd'},
      hostListeners:{keydown:'_keydown($event)',blur:'_blur()',focus:'_focus()'},hostProperties:{},
      hostAttributes:{class:'mat-chip-input mat-input-element'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdChipRemoveNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdChipRemove,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.MdChip}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[mdChipRemove], [matChipRemove]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{click:'_handleClick($event)'},
      hostProperties:{},hostAttributes:{class:'mat-chip-remove'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MdBasicChipNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdBasicChip,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'md-basic-chip, [md-basic-chip], mat-basic-chip, [mat-basic-chip]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-basic-chip'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
