// @angular/cdk/table/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/cdk/table';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '../../../common/common.ngsummary';
import * as i4 from './index.ngfactory';
export function CdkTableModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.CdkTableModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
          useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],
      modules:[{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.CdkTableModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.CdkTable},{reference:i0.CdkRowDef},{reference:i0.CdkCellDef},
          {reference:i0.CdkCellOutlet},{reference:i0.CdkHeaderCellDef},{reference:i0.CdkColumnDef},
          {reference:i0.CdkCell},{reference:i0.CdkRow},{reference:i0.CdkHeaderCell},
          {reference:i0.CdkHeaderRow},{reference:i0.CdkHeaderRowDef},{reference:i0.RowPlaceholder},
          {reference:i0.HeaderRowPlaceholder}],exportedPipes:([] as any[])},CdkTableNgSummary,
      CdkRowDefNgSummary,CdkCellDefNgSummary,CdkCellOutletNgSummary,CdkHeaderCellDefNgSummary,
      CdkColumnDefNgSummary,CdkCellNgSummary,CdkRowNgSummary,CdkHeaderCellNgSummary,
      CdkHeaderRowNgSummary,CdkHeaderRowDefNgSummary,RowPlaceholderNgSummary,HeaderRowPlaceholderNgSummary,
      i3.CommonModuleNgSummary];
}
export function CdkTableNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkTable,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.IterableDiffers}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.Renderer2}}},{isAttribute:true,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{value:'role'}}],lifecycleHooks:[0,
      1,4,5]},isComponent:true,selector:'cdk-table',exportAs:(null as any),inputs:{trackBy:'trackBy',
      dataSource:'dataSource'},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'cdk-table'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.CdkColumnDef}}],
          first:false,descendants:false,propertyName:'_columnDefinitions',read:(null as any)},
          {selectors:[{identifier:{reference:i0.CdkHeaderRowDef}}],first:true,descendants:true,
              propertyName:'_headerDefinition',read:(null as any)},{selectors:[{identifier:{reference:i0.CdkRowDef}}],
              first:false,descendants:false,propertyName:'_rowDefinitions',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i0.RowPlaceholder}}],first:true,
          descendants:true,propertyName:'_rowPlaceholder',read:(null as any)},{selectors:[{identifier:{reference:i0.HeaderRowPlaceholder}}],
          first:true,descendants:true,propertyName:'_headerRowPlaceholder',read:(null as any)}],
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i4.View_CdkTable_0,
      rendererType:i4.RenderType_CdkTable,componentFactory:i4.CdkTableNgFactory}];
}
export function CdkRowDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkRowDef,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.IterableDiffers}}}],lifecycleHooks:[3]},
      isComponent:false,selector:'[cdkRowDef]',exportAs:(null as any),inputs:{columns:'cdkRowDefColumns'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function CdkCellDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkCellDef,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[cdkCellDef]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function CdkCellOutletNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkCellOutlet,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ViewContainerRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[cdkCellOutlet]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function CdkHeaderCellDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkHeaderCellDef,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[cdkHeaderCellDef]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function CdkColumnDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkColumnDef,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[cdkColumnDef]',exportAs:(null as any),inputs:{name:'cdkColumnDef'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.CdkCellDef}}],
          first:true,descendants:true,propertyName:'cell',read:(null as any)},{selectors:[{identifier:{reference:i0.CdkHeaderCellDef}}],
          first:true,descendants:true,propertyName:'headerCell',read:(null as any)}],
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function CdkCellNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkCell,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.CdkColumnDef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'cdk-cell',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'cdk-cell',
          role:'gridcell'},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function CdkRowNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkRow,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'cdk-row',exportAs:(null as any),inputs:{},outputs:{},
      hostListeners:{},hostProperties:{},hostAttributes:{class:'cdk-row',role:'row'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i4.View_CdkRow_0,
      rendererType:i4.RenderType_CdkRow,componentFactory:i4.CdkRowNgFactory}];
}
export function CdkHeaderCellNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkHeaderCell,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.CdkColumnDef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'cdk-header-cell',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'cdk-header-cell',
          role:'columnheader'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function CdkHeaderRowNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkHeaderRow,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'cdk-header-row',exportAs:(null as any),inputs:{},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'cdk-header-row',
          role:'row'},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i4.View_CdkHeaderRow_0,
      rendererType:i4.RenderType_CdkHeaderRow,componentFactory:i4.CdkHeaderRowNgFactory}];
}
export function CdkHeaderRowDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkHeaderRowDef,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.IterableDiffers}}}],lifecycleHooks:[3]},
      isComponent:false,selector:'[cdkHeaderRowDef]',exportAs:(null as any),inputs:{columns:'cdkHeaderRowDef'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function RowPlaceholderNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.RowPlaceholder,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ViewContainerRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[rowPlaceholder]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function HeaderRowPlaceholderNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.HeaderRowPlaceholder,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ViewContainerRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[headerRowPlaceholder]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
